test/e2e/e2e.go

```diff
@@ -1241,32 +1241,68 @@ spec:
 		// create the pod that will be used as the source for the connectivity test
 		createSrcPod(srcPodName, serverNodeInfo.name, retryInterval, retryTimeout, f)
 
+		// In very rare cases the 'nc' test commands do not return the expected result on the first try,
+		// but while testing has reproduced these cases, the condition is temporary and only initially
+		// after the pod has been created. Eventually the traffic tests work as expected, so they
+		// are wrapped in PollImmediate with a total duration equal to 4 times the testTimeout
+		// which should provide for 3 tries before a failure is actually flagged
+		pollingDuration, _ := time.ParseDuration(testTimeout)
+		pollingDuration = pollingDuration * 4
+
 		// Verify the remote host/port as explicitly allowed by the firewall policy is reachable
 		ginkgo.By(fmt.Sprintf("Verifying connectivity to an explicitly allowed host %s is permitted as defined by the external firewall policy", exFWPermitTcpDnsDest))
-		_, err := framework.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "nc", "-vz", "-w", testTimeout, exFWPermitTcpDnsDest, "53")
+		err := wait.PollImmediate(1, pollingDuration, func() (bool, error) {
+			_, err := framework.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "nc", "-vz", "-w", testTimeout, exFWPermitTcpDnsDest, "53")
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
 		if err != nil {
 			framework.Failf("Failed to connect to the remote host %s from container %s on node %s: %v", exFWPermitTcpDnsDest, ovnContainer, serverNodeInfo.name, err)
 		}
+
 		// Verify the remote host/port as implicitly denied by the firewall policy is not reachable
 		ginkgo.By(fmt.Sprintf("Verifying connectivity to an implicitly denied host %s is not permitted as defined by the external firewall policy", exFWDenyTcpDnsDest))
-		_, err = framework.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "nc", "-vz", "-w", testTimeout, exFWDenyTcpDnsDest, "53")
-		if err == nil {
+		err = wait.PollImmediate(1, pollingDuration, func() (bool, error) {
+			_, err = framework.RunKubectl(f.Namespace.Name, "exec", srcPodName, testContainerFlag, "--", "nc", "-vz", "-w", testTimeout, exFWDenyTcpDnsDest, "53")
+			if err == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
```

Comment:
woops this almost tripped me up haha, i realised you have switched the err==nil within the pollImmediate, makes sense!
and these tests are passing in all the control-plane lanes so we are good.

Reply:
yeah, this took me a minute to figure out what was what.